src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java

```diff
@@ -35,16 +36,29 @@
 import hudson.tasks.Publisher;
 import hudson.tasks.Recorder;
 import jenkins.tasks.SimpleBuildStep;
+import org.kohsuke.stapler.DataBoundSetter;
 
 @SuppressWarnings({"unused", "WeakerAccess"})
 public class CompareCoverageAction extends Recorder implements SimpleBuildStep {
 
     private static final long serialVersionUID = 1L;
+    private String sonarLogin;
+    private String sonarPassword;
```

Comment:
Actually this exactly my case. I use this plugin from Jenkins pipelines. We use a permanent login and our password is a generated token with several hours validness. So I can only specify login in configuration, but I would still prefer to pass it from pipelines, because I created an internal library for pipelines and I can get this login on-the-fly, so for better user of my library experience I would prefer to pass it from pipelines.